macos/FluentUI/Button/Button.swift

```diff
@@ -366,20 +366,34 @@ open class Button: NSButton {
 // MARK: - ButtonCell
 
 class ButtonCell: NSButtonCell {
-	fileprivate var verticalPadding: CGFloat = 0
-	fileprivate var horizontalPadding: CGFloat = 0
-	fileprivate var titleVerticalPositionAdjustment: CGFloat = 0
-	fileprivate var titleToImageSpacing: CGFloat = 0
-	fileprivate var titleToImageVerticalSpacingAdjustment: CGFloat = 0
+	var verticalPadding: CGFloat = 0
+	var horizontalPadding: CGFloat = 0
+	var titleVerticalPositionAdjustment: CGFloat = 0
+	var titleToImageSpacing: CGFloat = 0
+	var titleToImageVerticalSpacingAdjustment: CGFloat = 0
+
+	let OnePointSize = NSSize(width: 1, height: 1)
 
 	override func imageRect(forBounds rect: NSRect) -> NSRect {
 		guard
 			let image = image,
 			let controlView = controlView,
-			image.size != NSZeroRect.size,
+			image.size != .zero,
 			imagePosition != .noImage
 		else {
-			return NSZeroRect
+			return .zero
+		}
+
+		// The test app exposed cases where a button with no image has its
+		// imagePosition set to .imageOnly or .imageOverlaps, then back to
+		// something else.  An invisible 1-pixel image gets assigned to the
+		// button, having an NSCustomImageRep with no drawing delegate.  Detect
+		// and remove this image so it doesn't offset the title.
```

Comment:
This feels like something we should file an Apple bug on? I think long term this seems like something that would break often as OS implementation changes so we should push to actually get the behavior fixed and maybe consider just not supporting that scenario if we don't expect it to be a common case so all usages of it don't get the weight/complexity they may not need

Reply:
Moved this into the test app code.  Can follow up with a minimal test project and Apple bug.